File: typescript/easy/greatest-common-divisor-of-strings.ts
function gcdOfStrings(str1: string, str2: string): string {
    function gcd(a: number, b: number): number {
        return b === 0 ? a : gcd(b, a % b);
    }

    const maxLength = gcd(str1.length, str2.length);

    if (str1 + str2 !== str2 + str1) {
        return "";
    }

    return str1.substring(0, maxLength);
}

File: csharp/easy/greatest-common-divisor-of-strings.cs
string GcdOfStrings(string str1, string str2) {
    int Gcd(int a, int b) {
        return b == 0 ? a : Gcd(b, a % b);
    }

    int maxLength = Gcd(str1.Length, str2.Length);

    if (str1 + str2 != str2 + str1) {
        return "";
    }

    return str1.Substring(0, maxLength);
}
